File: UIPeriodicTable/CPlusPlus/Resources/Components/code_uibutton.cpp
// UIButton application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_uibutton( UIWidget *layout)
{
    PODVector<UIWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp, HandleUibuttonEvent ));
}

void PeriodicApp::HandleUibuttonEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    UIWidget* widget = static_cast<UIWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uibuttoncode" )
        {
            AppLog( "UIButton support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Components/code_uibutton.cpp", widget->GetParent() );
        }
        if (widget->GetId() == "uibuttonlayout" )
        {
            AppLog( "UIButton support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_uibutton.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() == "demobutton" )
        {
            AppLog( "UIButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttonducky" )
        {
            AppLog( "UIButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttonready" )
        {
            AppLog( "UIButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttonatomic" )
        {
            AppLog( "UIButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttongreen" )
        {
            AppLog( "UIButton action : " + widget->GetId() + " was pressed ");
        }
    }
}

File: UIPeriodicTable/CSharp/Resources/Components/code_uibutton.cs
// UIButton application source code
using System;
using AtomicEngine;

public class code_uibutton : CSComponent {

    public void Setup( UIWidget layout )
    {
        var dest = new AtomicEngine.Vector<AtomicEngine.UIWidget>();
        layout.SearchWidgetClass( "TBButton", dest );
        for (var ii = 0; ii < dest.Size; ii++) {  // set bulk event handlers on all buttons -- boom!
            dest[ii].SubscribeToEvent<WidgetEvent> (dest [ii], HandleUibuttonEvent );
        }
    }

    private static void HandleUibuttonEvent( WidgetEvent ev )
    {
        UIWidget widget = (UIWidget)ev.Target;
        if ( widget.Equals(null)) return;
        if ( ev.Type == UI_EVENT_TYPE.UI_EVENT_TYPE_CLICK) {
            if (widget.GetId() == "uibuttoncode" ) {
                AtomicMain.AppLog( "UIButton support : " + widget.GetId() + " was pressed " );
                AtomicMain.ViewCode ( "Components/code_uibutton.cs", widget.GetParent() );
            }
            if (widget.GetId() == "uibuttonlayout" ) {
                AtomicMain.AppLog( "UIButton support : " + widget.GetId() + " was pressed ");
                AtomicMain.ViewCode ( "Scenes/layout_uibutton.ui.txt", widget.GetParent() );
            }

            if (widget.GetId() == "demobutton" ) {
                AtomicMain.AppLog( "UIButton action : " + widget.GetId() + " was pressed ");
            }
            if (widget.GetId() == "buttonducky" ) {
                AtomicMain.AppLog( "UIButton action : " + widget.GetId() + " was pressed ");
            }
            if (widget.GetId() == "buttonready" ) {
                AtomicMain.AppLog( "UIButton action : " + widget.GetId() + " was pressed ");
            }
            if (widget.GetId() == "buttonatomic" ) {
                AtomicMain.AppLog( "UIButton action : " + widget.GetId() + " was pressed ");
            }
            if (widget.GetId() == "buttongreen" ) {
                AtomicMain.AppLog( "UIButton action : " + widget.GetId() + " was pressed ");
            }
        }
    }
}

